From 51708b699ed35f20d7180b1ecb81c5e6fb71aee9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Iy=C3=A1n?= <me@iyanmv.com>
Date: Fri, 13 Dec 2024 23:08:35 +0100
Subject: [PATCH] Remove hardcoded build paths & modify basic workflow to build
 in random path (#2019)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

* Remove hardcoded build paths

This fixes #2018 using `helpers.get_current_build_dir_name()`

Signed-off-by: Iyán Méndez Veiga <me@iyanmv.com>

* Use a random build path in the basic build test

Signed-off-by: Iyán Méndez Veiga <me@iyanmv.com>

---------

Signed-off-by: Iyán Méndez Veiga <me@iyanmv.com>
---
 .github/workflows/basic.yml | 51 +++++++++++++++++++++----------------
 tests/test_acvp_vectors.py  | 15 +++++++----
 2 files changed, 39 insertions(+), 27 deletions(-)

diff --git a/tests/test_acvp_vectors.py b/tests/test_acvp_vectors.py
index ee45f8c12..28cd4614c 100644
--- a/tests/test_acvp_vectors.py
+++ b/tests/test_acvp_vectors.py
@@ -120,9 +120,10 @@ def test_acvp_vec_sig_keygen(sig_name):
                     seed = testCase["seed"]
                     pk = testCase["pk"]
                     sk = testCase["sk"]
-                    
+
+                    build_dir = helpers.get_current_build_dir_name()
                     helpers.run_subprocess(
-                        ['build/tests/vectors_sig', sig_name, "keyGen", seed, pk, sk]
+                        [f'{build_dir}/tests/vectors_sig', sig_name, "keyGen", seed, pk, sk]
                     )
 
         assert(variantFound == True)
@@ -146,8 +147,10 @@ def test_acvp_vec_sig_gen_deterministic(sig_name):
                     sk = testCase["sk"]
                     message = testCase["message"]
                     signature = testCase["signature"]
+
+                    build_dir = helpers.get_current_build_dir_name()
                     helpers.run_subprocess(
-                        ['build/tests/vectors_sig', sig_name, "sigGen_det", sk, message, signature]
+                        [f'{build_dir}/tests/vectors_sig', sig_name, "sigGen_det", sk, message, signature]
                     )
 
         assert(variantFound == True)
@@ -173,8 +176,9 @@ def test_acvp_vec_sig_gen_randomized(sig_name):
                     signature = testCase["signature"]
                     rnd = testCase["rnd"]
                     
+                    build_dir = helpers.get_current_build_dir_name()
                     helpers.run_subprocess(
-                        ['build/tests/vectors_sig', sig_name, "sigGen_rnd", sk, message, signature, rnd]
+                        [f'{build_dir}/tests/vectors_sig', sig_name, "sigGen_rnd", sk, message, signature, rnd]
                     )
 
         assert(variantFound == True)
@@ -200,8 +204,9 @@ def test_acvp_vec_sig_ver(sig_name):
                     signature = testCase["signature"]
                     testPassed = "1" if testCase["testPassed"] else "0"
                     
+                    build_dir = helpers.get_current_build_dir_name()
                     helpers.run_subprocess(
-                        ['build/tests/vectors_sig', sig_name, "sigVer", pk, message, signature, testPassed]
+                        [f'{build_dir}/tests/vectors_sig', sig_name, "sigVer", pk, message, signature, testPassed]
                     )
 
         assert(variantFound == True)
